definitions/scenarios/self_upgrade.rb

```diff
@@ -47,12 +47,12 @@ def req_repos_to_update_pkgs
       end
     end
 
-    def upstream_target_version
-      if feature(:katello_install)
-        return foreman_version_by_katello(target_version)
-      else
-        target_version
-      end
+    def self_upgrade_allowed?
+      Gem::Version.new(current_version) == Gem::Version.new(target_version)
```

Comment:
`current_version` is `X.Y.Z` while `target_version` is `X.Y`. So while this will work for `Z=0`, it won't for others.